vql/darwin/xattr.go

```diff
@@ -45,42 +42,46 @@ func (self XAttrFunction) Call(
 		return nil
 	}
 
-	self.Filename, err = accessors.GetUnderlyingAPIFilename(arg.Accessor, scope, arg.Filename)
-	self.Attributes = map[string]string{}
+	filename, err := accessors.GetUnderlyingAPIFilename(arg.Accessor, scope, arg.Filename)
 	if err != nil {
 		scope.Log("xattr: Failed to get underlying filename for %s: %s", arg.Filename.String(), err)
 	}
 
 	if len(arg.Attributes) > 0 {
-		self.getAttributeValues(scope, arg.Attributes)
+		attr = self.getAttributeValues(scope, arg.Attributes, filename)
 	} else {
-		attributes, err := xattr.List(self.Filename)
+		attributes, err := xattr.List(filename)
 		if err != nil {
-			scope.Log("xattr: Failed to list attributes for filename %s: %s", self.Filename, err)
-			return nil
+			scope.Log("xattr: Failed to list attributes for filename %s: %s", filename, err)
+			return vfilter.Null{}
 		}
 
-		self.getAttributeValues(scope, attributes)
+		attr = self.getAttributeValues(scope, attributes, filename)
 	}
 
-	if len(self.Attributes) == 0 {
-		return nil
+	if attr == nil {
+		return vfilter.Null{}
 	}
 
 	return ordereddict.NewDict().
-		Set("filename", self.Filename).
-		Set("attribute", self.Attributes)
+		Set("filename", filename).
```

Comment:
there is probably little value in echoing the same filename we just provided as an arg - so just return the attr dict directly